test/rpc_test.py

```diff
@@ -858,3 +859,13 @@ def test_func():
 
         if TEST_CONFIG.rpc_backend == RpcBackend.PROCESS_GROUP:
             self.assertEqual(test_func(), "expected result")
+
+    def test_process_group_agent_destructor(self):
+        dist.init_process_group(backend="gloo", init_method=self.init_method, timeout=timedelta(seconds=10))
+        rpc.init_model_parallel(
+            self_name="worker%d" % self.rank,
+            backend=TEST_CONFIG.rpc_backend,
+            self_rank=self.rank,
+            init_method=self.init_method,
+        )
+        # exception would be raised here if process group agent exited without detaching/joining threads.
```

Comment:
With your changes, an exception shouldn't be raised right?

Reply:
Yep, this comment was meant to indicate what would happen in the test if the process group agent was not destructed correctly. I'll make this comment more clear.